content/blog/understanding-the-dom/index.md

```diff
@@ -0,0 +1,519 @@
+
+> While you've seen `style` attribute be read and written to by an object interface, if you inspect the element or use the `getAttribute` to access the attribute's HTML API value, you'll find that it's really a string with a pleasant API wrapped around it that lets you use an object to interface with the attribute value.
+>
+> ```javascript
+> console.log(mainTextElement.getAttribute('style')); // This will return a string value, despite the API that lets you use an object to read and write
+> ```
+>
+> The reasoning behind this incongruity is due to [the implicit mapping of the HTML API and the `Element` API as mentioned at the start of the previous section](#element-properties). The limitations described here will also apply to the HTML API of those types of properties.
+
+For example, we can [use `data` attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) in order to read and write values via attributes to any given element.
+
+```html
+<!-- index.html -->
+<ul id="list" data-listitems="2">
+  <li>List item 1</li>
+  <li>List item 2</li>
+</ul>
+```
+
+```javascript
+// index.js
+const listEl = document.querySelector('#list');
+console.log(listEl.dataset.listitems); // '2'
+listEl.dataset.listitems = 3;
+console.log(listEl.dataset.listitems); // '3'
+```
+
+![Demonstrating that dataset values are able to be read and written](list_dataset.png)
+
+If you'll notice, I wrote the string `'3'` instead of the numerical value `3` in the code sample's outputs in the comments despite using the numerical `3` to set the value. This behavior is due to how default non-string values are saved to attributes.
+
+By default, the primitive's `toString` will be called to store values.
+
+```javascript
+element.dataset.userInfo = {name: "Tony"}
+console.log(element.dataset.userInfo) // "[object Object]"
+/**
+ * "[object Object]" is because it's running `Object.prototype.toString()`
+ * to convert the object to a string to store on the attribute
+ */
+```
+
+> If you're having a difficult time understanding why `toString` is bring ran or what `prototype` is doing here, don't worry: You're in good company. The JavaScript prototype system is complex and can be difficult to follow.
+>
+> For now, it will suffice to just know that you're only able to store strings in element attribute.
+
+
+
+## Events {#events}
+
+Just as your browser uses the DOM to know where to place elements to display for the user and metadata about the elements in order to interact with screenreaders as-expected, your browser also utilizes the DOM for knowing how to handle user interactions. The way your browser handles user interaction is by listening for _events_ that occur when the user takes action or other noteworthy changes occur.
```

Comment:
This is extremely wordy and should be split into multiple sentences if everything is needed.

Reply:
```suggestion
Just as your browser uses the DOM to handle on-screen content visibility, your browser also utilizes the DOM for knowing how to handle user interactions. The way your browser handles user interaction is by listening for _events_ that occur when the user takes action or other noteworthy changes occur.
```